rust/src/lib/policy/token.rs

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use std::iter::FromIterator;
+
+use crate::{ErrorKind, NmstateError};
+
+const NEXT_TOKEN_START_CHARS: [char; 6] = [' ', '=', '!', '|', '"', ':'];
+
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub(crate) enum NetworkCaptureToken {
```

Comment:
Path is too big for a token and Value can be more concrete, we can do something like 

```rust
pub(crate) enum Token {
    Identity(usize, String), // foo
    Number(usize, i32), // 23
    Str(usize, String), // "foo"
    Dot(usize),      // .
    Pipe(usize),     // |
    Replace(usize),  // !=
    Filter(usize), // ==
}
```

Then we construct Path later on as a vector of enums as part of the commands tree.

Reply:
If we do that, we are introducing complexity for no reason. Why twice?

Reply:
Ok at least lest identify at tokenize between String, Number and Identity.